test/unit/Github/JwageGenerateChangelogTest.php

```diff
@@ -32,7 +41,22 @@ public function testGenerateChangelog(): void
         $repositoryName = RepositoryName::fromFullName('laminas/repository-name');
         $semVerVersion  = SemVerVersion::fromMilestoneName('1.0.0');
 
-        (new JwageGenerateChangelog($changelogGenerator))
+        (new JwageGenerateChangelog($changelogGenerator, $githubCredentials))
             ->__invoke($repositoryName, $semVerVersion);
     }
+
+    public function testCreateGenerateChangelog(): void
+    {
+        $makeRequests = $this->createMock(RequestFactoryInterface::class);
+        $httpClient   = $this->createMock(HttpClient::class);
+        $githubToken  = $this->createMock(GitHubCredentials::class);
+
+        $changeLogGenerator = JwageGenerateChangelog::create(
+            $makeRequests,
+            $httpClient,
+            $githubToken
+        );
+
+        $this->assertInstanceOf(JwageGenerateChangelog::class, $changeLogGenerator);
```

Comment:
This isn't really testing anything (the type declarations already do this for us): is there a way for us to somehow verify that the credentials were set?

Reply:
Yes, during the __invoke call, I'll add the test.

Reply:
@Ocramius The token is used only [here](https://github.com/Lansoweb/automatic-releases/blob/private-repository/src/Github/JwageGenerateChangelog.php#L54) to create a ```ChangelogConfig``` to be used by the ```JwageGenerateChangelog```.

Reply:
Yeh, it's just that the test as-is will surely be marked as invalid in future iterations of the psalm PHPUnit plugin, because there is a 100% type match here, so the assertion will be detected as redundant.

In practice I'm OK with this patch as-is, and to improve the test in a later iteration.